doc/page-properties.h

```diff
@@ -0,0 +1,129 @@
+/**********************************************************************
+**
+** Copyright (C) Luxoft Sweden AB 2020
+**
+** This file is part of the documentation of the FaceLift project
+**
+** This work is licensed under the Creative Commons
+** Attribution-ShareAlike 4.0 International License. To view a copy of
+** this license, visit http://creativecommons.org/licenses/by-sa/4.0/
+** or send a letter to Creative Commons, PO Box 1866, Mountain View,
+** CA 94042, USA.
+**
+** SPDX-License-Identifier: CC-BY-SA-4.0
+**
+**********************************************************************/
+
+/**
+\page page-properties Properties
+
+\section properties-intro Introduction
+Properties are the part of the interface definition described in *.qface file. Eventually they will be generated as C++ and QML properties of the interface. For more details about properties in *.qface files see documentation at https://github.com/Pelagicore/qface
+
+In Facelift each property is exposed to the QML engine as Q_PROPERTY with a setter and getter functions, and a change signal. If the property is marked as 'readonly' then no setter is generated.
+
+The facelift::Property class template provides encapsulation for the internal data with a const getter only (except for the service property), which guarantees that the data will not be unnoticibly changed. Each time the setter is called, with a value that differs from the previous one, the change signal is emitted. If the previous value equals to the new value, the change signal will not be emitted.
+
+\section QML
+
+The property is exposed directly to the QML engine by Q_PROPERTY macro and can be used without any additional effort.
```

Comment:
are you talking about implementing an interface using QML ? Please make it clear.

Reply:
I am talking about property exposure to QML engine, not QML implementation. Will be rephrased.